### DISCUSSION TITLE
[v2] Display an image using the asset protocol.

### DISCUSSION THREAD

**Discussion by charmbyte** [tauri-apps/tauri#11498] [Q&A] [answered]
I am trying to display an image using the asset protocol, but I continue to get the error: 403 Forbidden. This is what I have in my tauri.conf.json:

```
"security": {
      "csp": "default-src 'self' ipc: http://ipc.localhost; img-src 'self' asset: http://asset.localhost",
      "assetProtocol": {
        "enable": true,
        "scope": ["*/**"]
      }
    }
```

I then followed the example located here: https://tauri.app/reference/javascript/api/namespacecore/#convertfilesrc. If I console.log my assetUrl path, it seems to work. It logs something like "asset://....../image.jpg".

Is there a step that I'm missing?

**Comment by FabianLars** [comment]
Try `"**/*"` instead, or use the dialog api to select the path which will expand the scope at runtime.

**Comment by charmbyte** [comment]
@FabianLars I should mention that I primarily develop on Arch Linux. I couldn't get the images to load no matter what process I used -- that is until I switch to Windows for development temporarily. Then suddenly the exact same code would show the images and I wouldn't get any 403 Forbidden errors. So I went back to Linux just to be sure, and sure enough I would get the 403 Forbidden error. Does using the asset protocol in this way only work for Windows?

Thanks!

**Reply by FabianLars** [reply]
No, for images the asset protocol should work on all platforms. Only audio/video has issues on linux.

**Reply by charmbyte** [reply]
Yea, I've found issues and discussions regarding the audio/video issues. I just can't figure out why the same code would work on Windows but not Linux. The image I'm loading is in the .local/share folder (app_data_dir). Could it be a permissions issue of some kind on Linux? Regardless, I'll keep digging and see if there's something I'm missing.

**Reply by FabianLars** [reply, marked as answer]
yes, it can. i'm not sure how the asset protocol scope is implemented but for the fs scope you have to explicitly add the "hidden" folders (the ones starting with `.`) in the scope paths unless the plugin is configured otherwise via `requireLiteralLeadingDot: false`

Edit: This would not be required if you'd use the predefined scope variables btw

**Reply by charmbyte** [reply]
Sure enough, adding `**/.local/share/**/*` to the scope seems to work. Thank you for the help!

**Reply by sklymoshenko** [reply]
> yes, it can. i'm not sure how the asset protocol scope is implemented but for the fs scope you have to explicitly add the "hidden" folders (the ones starting with `.`) in the scope paths unless the plugin is configured otherwise via `requireLiteralLeadingDot: false`
> 
> Edit: This would not be required if you'd use the predefined scope variables btw


~~Hi, I am using a $CACHE to store some images, and giving a permissions as~~
```
 "security": {
      "assetProtocol": {
        "scope": ["$CACHE/**", "**/.cache/**"],
        "enable": true
      }
}
```
~~still getting a 403 forbidden when trying to access it through assets://localhost. It used to work just fine on v1.~~
~~And looks like on macOS it's not an issue, but Ubuntu 24 has it.~~

Update:
I am using multiple configs for different OS, it was rewritten in `tauri.linux.conf.json` which had only `scope: [**]` which doesnt includes a hidden folders.
